plugins/webp-uploads/settings.php

```diff
@@ -210,7 +210,19 @@ function webp_uploads_use_picture_element_callback(): void {
 	</div>
 	<div id="webp_uploads_picture_element_fieldset" class="<?php echo ! $jpeg_fallback_enabled ? 'disabled' : ''; ?>">
 		<label for="webp_uploads_use_picture_element" id="webp_uploads_use_picture_element_label">
-			<input name="webp_uploads_use_picture_element" type="checkbox" id="webp_uploads_use_picture_element" aria-describedby="webp_uploads_use_picture_element_description" value="1"<?php checked( webp_uploads_is_picture_element_enabled() ); ?> class="<?php echo ! $jpeg_fallback_enabled ? 'disabled' : ''; ?>" >
+			<input
+				type="checkbox"
+				id="webp_uploads_use_picture_element"
+				aria-describedby="webp_uploads_use_picture_element_description"
+				<?php checked( get_option( 'webp_uploads_use_picture_element', false ) ); // Option intentionally used instead of webp_uploads_is_picture_element_enabled() to persist when perflab_generate_webp_and_jpeg is updated. ?>
+				<?php disabled( ! $jpeg_fallback_enabled ); ?>
+				onchange="document.getElementById('webp_uploads_use_picture_element_value').value = this.checked ? 1 : 0"
+			>
```

Comment:
Does the lack of a `name` attribute on this have any unintended side effects? Makes sense we don't send it with the form submission in favor of the hidden input, but I wonder if this leads to other problems (is the `name` attribute used for anything else?).

Alternatively, we avoid tampering with this (user-facing) element completely and inject the hidden input only when needed (if the checkbox is disabled). This could then happen in the above event listener for `#perflab_generate_webp_and_jpeg`, which IMO is cleaner in terms of code organization too as it colocates all the dynamic logic in one place based on the condition (rather than the `onchange` attribute here that feels a bit out of place).

Reply:
> Does the lack of a `name` attribute on this have any unintended side effects? Makes sense we don't send it with the form submission in favor of the hidden input, but I wonder if this leads to other problems (is the `name` attribute used for anything else?).

No, AFAIK the `name` attribute is only relevant here in that it includes the value in the form submission.

> Alternatively, we avoid tampering with this (user-facing) element completely and inject the hidden input only when needed (if the checkbox is disabled). This could then happen in the above event listener for `#perflab_generate_webp_and_jpeg`, which IMO is cleaner in terms of code organization too as it colocates all the dynamic logic in one place based on the condition (rather than the `onchange` attribute here that feels a bit out of place).

IMO, adding and removing a hidden input makes the code more complicated. But if you'd like to see that, feel free to amend the PR.

Reply:
IMO it only makes the code _more_, but not more complicated. Happy to add this.